src/pages/studyView/resources/FilesAndLinks.tsx

```diff
@@ -112,13 +127,47 @@ async function fetchFilesLinksData(
         0
     );
 
-    const resourcesForPatients = await getResourceDataOfPatients(
+    const selectedSamplesList = selectedSamples.map(item => item.sampleId);
+    const selectedPatientsList = [
+        ...new Set(selectedSamples.map(item => item.patientId)),
+    ];
+
+    const resourcesForEntireStudy = await getResourceDataOfEntireStudy(
         studyClinicalDataResponse.data
     );
-    const resourcesForPatientsAndSamples: { [key: string]: ResourceData[] } = {
-        ...sampleIdResourceData,
-        ...resourcesForPatients,
-    };
+
+    const resourcesForPatientsAndSamples = resourcesForEntireStudy
+        .filter(
+            (resource: ResourceData) =>
+                selectedSamplesList.includes(resource.sampleId) ||
+                selectedPatientsList.includes(resource.patientId)
+        )
+        .reduce(
+            (
+                idMap: { [key: string]: ResourceData[] },
+                resource: ResourceData
+            ) => {
+                const { resourceType } = resource.resourceDefinition;
```

Comment:
this whole operation would be more clearly handled by lodash's keyBy function.  we like to use these kind of named helpers because they make the operation more explicit.

Reply:
Hm, I couldn't make lodash keyBy function to work the way I wanted it to for this use case. In python this would be a case for collections.defaultDict. But I couldn't make this work where I could create new arrays and append to them.

LMK if you want me to work on this part some more.

Reply:
@hweej sorry, i mean _.groupBy

Reply:
I have replaced these lines with a single lodash groupBy('patientId'). Thanks @alisman !